### PR TITLE
Update ScaleWorkload to provide replicas as string & int value

### PR DESCRIPTION
**Pull request by SupriyaKasten**
## Change Overview

ScaleWorkload only allowed static int values for `replicas` parameter. With these changes, we can provide string values to `replicas` in addition to int, kanister will take care of conversion.

## Pull request type

Please check the type of change your PR introduces:
- [ ] Work in Progress
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Trival/Minor
- [ ] Bugfix
- [x] Feature
- [ ] Documentation

## Test Plan

- [ ] Manual
- [x] Unit test
- [ ] E2E

### CODE REVIEW

File: pkg/function/scale_workload.go [hunk 1]
```diff
@@ -59,11 +60,19 @@ func (*scaleWorkloadFunc) RequiredArgs() []string {
...
 
 func getArgs(tp param.TemplateParams, args map[string]interface{}) (namespace, kind, name string, replicas int32, err error) {
-	err = Arg(args, ScaleWorkloadReplicas, &replicas)
+	var rep interface{}
+	err = Arg(args, ScaleWorkloadReplicas, &rep)
 	if err != nil {
 		return namespace, kind, name, replicas, err
 	}
-
+	if val, ok := rep.(int); ok {
```
Comment (tdmanv) [inline pkg/function/scale_workload.go]: please use a type switch here: https://tour.golang.org/methods/16

File: pkg/function/scale_workload.go [hunk 2]
```diff
@@ -59,11 +60,19 @@ func (*scaleWorkloadFunc) RequiredArgs() []string {
...
+		replicas = int32(val)
+	} else if val, ok := rep.(string); ok {
+		strToInt := intstr.Parse(val)
```
Comment (tdmanv) [inline pkg/function/scale_workload.go]: let's just use https://golang.org/pkg/strconv/#Atoi

File: pkg/function/scale_workload.go [hunk 3]
```diff
@@ -59,11 +60,19 @@ func (*scaleWorkloadFunc) RequiredArgs() []string {
...
+		replicas = strToInt.IntVal
+	} else {
+		return namespace, kind, name, replicas, errors.Wrapf(err, "Failed to decode arg `%s`", ScaleWorkloadReplicas)
```
Comment (tdmanv) [inline pkg/function/scale_workload.go]: typically I'd use a naked return with named return values. 
```suggestion
		err = errors.Errorf("Invalid arg type %T for Arg %s ", rep, ScaleWorkloadReplicas)
		return
```

File: pkg/function/scale_test.go [hunk 4]
```diff
@@ -269,7 +269,7 @@ func (s *ScaleSuite) TestGetArgs(c *C) {
 				},
 			},
 			args: map[string]interface{}{
-				ScaleWorkloadReplicas: 2,
+				ScaleWorkloadReplicas: "2",
```
Comment (tdmanv) [inline pkg/function/scale_test.go]: Let's leave one of these an int to test both codepaths

Comment (SupriyaKasten) [reply pkg/function/scale_test.go]: @tdmanv we do have two of these as int.

File: pkg/function/scale_workload.go [hunk 5]
```diff
@@ -59,11 +60,27 @@ func (*scaleWorkloadFunc) RequiredArgs() []string {
...
 
 func getArgs(tp param.TemplateParams, args map[string]interface{}) (namespace, kind, name string, replicas int32, err error) {
-	err = Arg(args, ScaleWorkloadReplicas, &replicas)
+	var rep interface{}
+	err = Arg(args, ScaleWorkloadReplicas, &rep)
 	if err != nil {
 		return namespace, kind, name, replicas, err
 	}
 
+	var val int
```
Comment (pavannd1) [inline pkg/function/scale_workload.go]: [minor] Should this be a function? Okay to be done as an enhancement later

**Pull request merged**